File: mycollab-web/src/main/java/com/esofthead/mycollab/module/file/view/FileUrlResolver.java
package com.esofthead.mycollab.module.file.view;

import com.esofthead.mycollab.eventmanager.EventBus;
import com.esofthead.mycollab.module.crm.view.CrmUrlResolver;
import com.esofthead.mycollab.shell.events.ShellEvent;
import com.esofthead.mycollab.vaadin.mvp.ModuleHelper;
import com.esofthead.mycollab.vaadin.mvp.UrlResolver;

public class FileUrlResolver extends UrlResolver {

	public UrlResolver build() {
		this.addSubResolver("list", new FileListUrlResolver());
		return this;
	}

	@Override
	public void handle(String... params) {
		if (!ModuleHelper.isCurrentFileModule()) {
			EventBus.getInstance().fireEvent(
					new ShellEvent.GotoFileModule(this, params));
		} else {
			super.handle(params);
		}
	}

	@Override
	protected void defaultPageErrorHandler() {

	}

	public static class FileListUrlResolver extends CrmUrlResolver {

		@Override
		protected void handlePage(String... params) {
			// EventBus.getInstance().fireEvent(
			// new ShellEvent.GotoFileModule(this, null));
		}
	}

}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/file/view/FileUrlResolver.scala
package com.esofthead.mycollab.module.file.view

import com.esofthead.mycollab.eventmanager.EventBusFactory
import com.esofthead.mycollab.module.crm.view.CrmUrlResolver
import com.esofthead.mycollab.shell.events.ShellEvent
import com.esofthead.mycollab.vaadin.desktop.ui.ModuleHelper
import com.esofthead.mycollab.vaadin.mvp.UrlResolver

/**
 * @author MyCollab Ltd
 * @since 5.0.9
 */
class FileUrlResolver extends UrlResolver {
    def build: UrlResolver = {
        this.addSubResolver("list", new FileListUrlResolver)
        this
    }

    override def handle(params: String*) {
        if (!ModuleHelper.isCurrentFileModule) {
            EventBusFactory.getInstance.post(new ShellEvent.GotoFileModule(this, params))
        }
        else {
            super.handle(params: _*)
        }
    }

    protected def defaultPageErrorHandler {
    }

    class FileListUrlResolver extends CrmUrlResolver {
        protected override def handlePage(params: String*) {
        }
    }

}
